### DISCUSSION TITLE
traversal with Embedded not working

### DISCUSSION THREAD

**Discussion by Ranjith2002** [JanusGraph/janusgraph#3634] [Q&A]
gremlin> g=traversal().withEmbedded(graph)

No signature of method: org.apache.tinkerpop.gremlin.process.traversal.AnonymousTraversalSource.withEmbedded() is applicable for argument types: (org.janusgraph.graphdb.database.StandardJanusGraph) values: [standardjanusgraph[cql:[127.0.0.1]]]

Type ':help' or ':h' for help.

Display stack trace? [yN]

what i need to do? then what is difference from g=graph.traversal()

**Comment by hadoopmarc** [comment]
OK, I do the following in the janusgraph 0.6.2. binary distribution:
```
$ bin/gremlin.sh
...
gremlin> graph = JanusGraphFactory.open('conf/janusgraph-inmemory.properties')
==>standardjanusgraph[inmemory:[127.0.0.1]]
gremlin> g = traversal().withEmbedded(graph) 
==>graphtraversalsource[standardjanusgraph[inmemory:[127.0.0.1]], standard]
gremlin> 
```
In what way is your start situation different?

Marc

**Comment by Ranjith2002** [comment]
I am using gremlin version 0.5.3
In that only gremlin-server.sh file there.

Other versions don't have that file.so that kind of versions I can't start
my gremlin server

Ok sir but why that kind of traversal we are using?

Difference from g.traversal()

version 0.6.2 doesn't have gremlin-server.sh,  how i start the server
![aa](https://user-images.githubusercontent.com/105899472/222051248-3631c4d3-22d7-4b75-a365-0a9031c32936.PNG)

**Comment by hadoopmarc** [comment]
You can download janusgraph from here: https://github.com/JanusGraph/janusgraph/releases
The withEmbedded() construct is for obtaining a GraphTraversalSource of a local graph instance. For connecting to the Gremlin Server of JanusGraph you have to use the withRemote() construct, see: https://tinkerpop.apache.org/docs/current/reference/#gremlin-java-connecting

Marc

**Comment by Ranjith2002** [comment]
sir in janusgraph 0.6.2 file doesn't contain gremlin-server.sh (bin folder)
. then how i can able to run gremlin server

attached here photo
[image: aa.PNG]

**Comment by hadoopmarc** [comment]
The JanusGraph bin folder has two relevant scipts:

1. bin/janusgraph.sh start
2. bin/janusgraph-server.sh start

The first starts, in this order, a local Cassandra instance, a local Elasticsearch instance and a local Gremlin-Server instance.
The second script only starts a local Gremlin-Server instance with the inmemory storage backend.

So, the JanusGraph distribution includes the TinkerPop Gremlin-Server and all TinkerPop documentation applies also to JanusGraph.

Marc
